src/api/ums/umsModule.ts

```diff
@@ -0,0 +1,81 @@
+import { ReflectiveInjector } from "injection-js";
+import { API } from "../../config";
+import { Methods, RequestAdapter } from "../../internal/requestAdapter";
+import { Client, ClientInit } from "../core/client";
+import { IModule } from "../core/module";
+import { AuthOptions, TokenManager } from "../core/tokenManager";
+
+export interface IUMSSignInOptions {
+  email: string;
+  password: string;
+  default?: boolean;
+  auth?: string;
+}
+
+export type IUMSCredentials = Pick<IUMSSignInOptions, 'email' | 'password'>;
+
+export interface IUMSUser {
+  id: string;
+  email: string;
+}
+
+export class UMSModule implements IModule {
+
+  private tokenManager: TokenManager;
+
+  private requestAdapter: RequestAdapter;
+
+  private systemInit: Promise<Client>;
+
+  private projectId: string;
+
+  public init(coreInjector: ReflectiveInjector) {
```

Comment:
Please add an explicit return:
```suggestion
  public init(coreInjector: ReflectiveInjector): Promise<UMSModule> {
```

Reply:
@marlosin Actually it's not needed because of
```
export class UMSModule implements IModule {
```
and
```
/**
 * Interface used for all Jexia modules
 */
export interface IModule {
  /**
   * Initialize the Jexia Module
   * @param coreInjector main dependency injector of the Jexia client
   */
  init(coreInjector: ReflectiveInjector): Promise<this>;
  /**
   * Terminate the Jexia Module
   */
  terminate(): Promise<this>;
}
```